client/ayon_core/tools/sceneinventory/view.py

```diff
@@ -735,45 +772,72 @@ def _show_version_dialog(self, item_ids, active_repre_id):
+
+        # active_project_name = None
+        active_repre_info = None
+        repre_info_by_project = {}
+        version_items_by_project = {}
+        for project_name, repre_ids in repre_ids_by_project.items():
+            repres_info = self._controller.get_representation_info_items(
+                project_name, repre_ids
+            )
+            if active_repre_info is None:
+                # active_project_name = project_name
+                active_repre_info = repres_info.get(active_repre_id)
+
+            product_ids = {
+                repre_info.product_id
+                for repre_info in repres_info.values()
+                if repre_info.is_valid
+            }
+            version_items_by_product_id = self._controller.get_version_items(
+                project_name, product_ids
+            )
+
+            repre_info_by_project[project_name] = repres_info
+            version_items_by_project[project_name] = (
+                version_items_by_product_id
+            )
 
-        product_ids = {
-            repre_info.product_id
-            for repre_info in repre_info_by_id.values()
-        }
-        active_repre_info = repre_info_by_id[active_repre_id]
         active_version_id = active_repre_info.version_id
-        active_product_id = active_repre_info.product_id
-        version_items_by_product_id = self._controller.get_version_items(
-            product_ids
-        )
-        version_items = list(
-            version_items_by_product_id[active_product_id].values()
-        )
-        versions = {version_item.version for version_item in version_items}
-        product_ids_by_version = collections.defaultdict(set)
-        for version_items_by_id in version_items_by_product_id.values():
-            for version_item in version_items_by_id.values():
-                version = version_item.version
-                _prod_version = version
-                if _prod_version < 0:
-                    _prod_version = -1
-                product_ids_by_version[_prod_version].add(
-                    version_item.product_id
-                )
-                if version in versions:
-                    continue
-                versions.add(version)
-                version_items.append(version_item)
+        # active_product_id = active_repre_info.product_id
 
-        def version_sorter(item):
+        versions = set()
+        product_ids = set()
+        version_items = []
+        product_ids_by_version_by_project = {}
+        for project_name, version_items_by_product_id in (
+            version_items_by_project.items()
+        ):
+            product_ids_by_version = collections.defaultdict(set)
+            product_ids_by_version_by_project[project_name] = (
+                product_ids_by_version
+            )
+            for version_item_by_id in version_items_by_product_id.values():
+                for version_item in version_item_by_id.values():
```

Comment:
The issue is fixed, but, I'm not sure if it fixes the problem or it's just a workaround.
out of curiosity I printed `version_items_by_product_id` 
and I found out that the version id is duplicated. is it expected? 

```python
{
    '98ba2c91a8e311efabcacff42558f871': {
        '98bc68c1a8e311efb681cff42558f871': <ayon_core.tools.sceneinventory.models.containers.VersionItem object at 0x000000013B2875D0>,
        '9df32065a8e311ef9b1ccff42558f871': <ayon_core.tools.sceneinventory.models.containers.VersionItem object at 0x0000000F82DC3D10>, 
        'c7310386a8e311efb146f02f74db125e': <ayon_core.tools.sceneinventory.models.containers.VersionItem object at 0x000000013EC05C90>}
}
```

Reply:
yeah, it makes sense because the first one refers to the version id of the  latest publish and dictionary refers to every version in that product.